### PR TITLE
Add "JSONGroup" and remove dependence on loader from new Nexus API

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- Add JSONGroup (named _JSONGroup until the small legacy of the same name is removed).
- Remove need to track the "loader" in NXobject, Field, and friends.

This is the last but one big step in removing the older LoadFromHdf5 and LoadFromJson classes. There should now be a fair amount of code that can be moved and removed. Next, _JSONGroup and JSONDataset will be refactored to avoid the cumbersome reliance on LoadFromJson. I chose to not include this refactoring step (which will result in code removal) here, since I felt that the diff would be much harder to decipher. This implies that a number of details in this PR are temporary.

### CODE REVIEW

File: src/scippneutron/file_loading/nxobject.py [hunk 1]
```diff
@@ -6,11 +6,12 @@
...
 import functools
 from typing import List, Union, NoReturn, Any, Dict, Tuple
+import numpy as np
 import scipp as sc
 import h5py
 
-from ._nexus import LoadFromNexus
-from ._hdf5_nexus import LoadFromHdf5, _cset_to_encoding, _ensure_str
+from ._hdf5_nexus import _cset_to_encoding, _ensure_str
+from ._hdf5_nexus import _ensure_supported_int_type, _warn_latin1_decode
```
Comment (jl-wynen) [inline src/scippneutron/file_loading/nxobject.py:14]: Importing a lot of protected symbols. Are you planning to factor their use out of this file?

File: src/scippneutron/file_loading/_json_nexus.py [hunk 2]
```diff
@@ -287,38 +287,107 @@ def __contains__(self, name):
...
 
-class JSONDataset:
-    def __init__(self, node: dict, loader: LoadFromJson):
+class JSONNode:
+    def __init__(self, node: dict, loader: LoadFromJson, *, parent=None):
+        self._file = parent.file if parent is not None else self
+        self._parent = self if parent is None else parent
         self._node = node
         self._loader = loader
+        print(self.name, parent is None, self._file.name, self._parent.name)
```
Comment (jl-wynen) [inline src/scippneutron/file_loading/_json_nexus.py:300]:
```suggestion
```

File: tests/nexus_test.py [hunk 3]
```diff
@@ -61,15 +70,15 @@ def builder_with_events_monitor_and_log():
 def test_nxobject_root(nexus_group: Tuple[Callable, LoadFromNexus]):
     resource, loader = nexus_group
```
Comment (jl-wynen) [inline tests/nexus_test.py:71]: Looks like `loader` can be removed from all those tests

File: src/scippneutron/file_loading/nxobject.py (hunk 1 above)
Comment (SimonHeybrock) [reply src/scippneutron/file_loading/nxobject.py:14]: Yes, this will all move.

File: tests/nexus_test.py (hunk 3 above)
Comment (SimonHeybrock) [reply tests/nexus_test.py:71]: Working on it in follow-up.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**